### PR TITLE
Download kustomize during manifests generation

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/17198

### CODE REVIEW

File: bootstrap/aws-china-test.yaml [hunk 1]
```diff
@@ -35,15 +35,20 @@ spec:
         description: Application is a definition of Application resource.
         properties:
           apiVersion:
-            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
```
Comment (kopiczko) [inline bootstrap/aws-china-test.yaml:40]: Apparently this version of `kustomize` doesn't like long strings.

File: Makefile.custom.mk [hunk 2]
```diff
@@ -1,33 +1,55 @@
...
+manifests/bases/upstream-argocd: ARGOCD_REPOSITORY := https://github.com/argoproj/argo-cd.git
+manifests/bases/upstream-argocd: WORK_DIR := $(shell mktemp -d -t ci-XXXXXXXXXX)
+manifests/bases/upstream-argocd: $(KUSTOMIZE)
 	@echo "====> $@"
-	./hack/update-argocd-manifests.sh $(ARGOCD_VERSION)
+	git -c advice.detachedHead=false clone --quiet --depth 1 --branch ${ARGOCD_VERSION} ${ARGOCD_REPOSITORY} ${WORK_DIR}
+	mkdir -p manifests/bases/upstream-argocd
+	echo "$(AUTOGENMSG)" > manifests/bases/upstream-argocd/install.yaml
+	$(KUSTOMIZE) build "$(WORK_DIR)/manifests/cluster-install" >> manifests/bases/upstream-argocd/install.yaml
+	rm -rf $(WORK_DIR)
```
Comment (kopiczko) [inline Makefile.custom.mk:20]: This is the hack script converted into a make target.

File: Makefile.custom.mk [hunk 3]
```diff
@@ -1,33 +1,55 @@
...
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+define go-get-tool
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+rm -rf $$TMP_DIR
+endef
```
Comment (kopiczko) [inline Makefile.custom.mk:55]: This downloads kustomize binary into bin/kustomize if it doesn't exist.

**Review by corest — APPROVED** [review]
I don't like makefiles, but won't fight for bash scripts

**Comment by kopiczko** [comment]
Fro me Makefiles have an edge when it comes to several things:

- They print the executed command by default. It's easy to opt out for a single command.
- It's easy to create a separate target and keep the common functionality.
- By default they fial on non-zero output.

But they definitely aren't perfect.

File: Makefile.custom.mk (hunk 3 above)
Comment (kopiczko) [reply Makefile.custom.mk:55]: BTW it is bash :D

**Pull request merged**